acceptance/tests/provider/package/yum.rb

```diff
@@ -0,0 +1,96 @@
+        assert_match(/\s#{package}/, list)
+      end
+      on(agent, "#{yum_command} remove #{package} -y")
+    end
+
+    step "Uninstall a yum package" do
+      on(agent, "#{yum_command} install #{package} -y")
+      package_manifest = resource_manifest('package', package, { ensure: 'absent', provider: 'yum' } )
+      apply_manifest_on(agent, package_manifest, :catch_failures => true) do
+        list = on(agent, "#{yum_command} list installed").stdout
+        assert_no_match(/\s#{package}/, list)
+      end
+    end
+
+    step "Downgrading a yum package" do
+      package_manifest = resource_manifest('package', package, { ensure: higher_package_version, provider: 'yum' } )
+      apply_manifest_on(agent, package_manifest, :catch_failures => true) do
+        list = on(agent, "#{yum_command} list installed").stdout
+        assert_match(/#{higher_package_version}/, list)
+      end
+      package_manifest = resource_manifest('package', package, { ensure: lower_package_version, provider: 'yum' } )
+      apply_manifest_on(agent, package_manifest, :catch_failures => true) do
+        list = on(agent, "#{yum_command} list installed").stdout
+        assert_match(/#{lower_package_version}/, list)
+      end
+      on(agent, "#{yum_command} remove #{package} -y")
+    end
+
+    step "Using semantic versioning to downgrade to a desired version < X" do
+      on(agent, "#{yum_command} install #{package} -y")
+      package_manifest = resource_manifest('package', package, { ensure: "<=#{lower_package_version}", provider: 'yum' } )
+      apply_manifest_on(agent, package_manifest, :catch_failures => true) do
+        list = on(agent, "#{yum_command} list installed").stdout
+        assert_match(/#{lower_package_version}/, list)
+      end
+      # idempotency test
+      package_manifest = resource_manifest('package', package, { ensure: "<=#{lower_package_version}", provider: 'yum' } )
+      exit_code = apply_manifest_on(agent, package_manifest, :catch_changes => true, :acceptable_exit_codes => [ 0 , 2 ]).exit_code
+      assert_equal(exit_code, 0)
+      on(agent, "#{yum_command} remove #{package} -y")
+    end
+
+    step "Using semantic versioning to ensure a version >= X" do
+      on(agent, "#{yum_command} install #{package} -y")
+      package_manifest = resource_manifest('package', package, { ensure: ">#{lower_package_version}", provider: 'yum' } )
+      exit_code = apply_manifest_on(agent, package_manifest, :catch_changes => true, :acceptable_exit_codes => [ 0 , 2 ]).exit_code
+      assert_equal(exit_code, 0)
+      on(agent, "#{yum_command} remove #{package} -y")
+    end
+
+    step "Using semantic versioning to install a version >= X" do
+      package_manifest = resource_manifest('package', package, { ensure: ">=#{higher_package_version}", provider: 'yum' } )
+      exit_code = apply_manifest_on(agent, package_manifest, :catch_changes => true, :acceptable_exit_codes => [ 0 , 2 ]).exit_code
```

Comment:
Here you can use `:expect_changes` instead of `:catch_changes` which allows only exit code 2. That way you don't need to get and assert the exit code separately.